zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/remote/ExecRemoteInterpreterProcess.java

```diff
@@ -248,6 +252,11 @@ public void onProcessFailed(ExecuteException e) {
       synchronized (this) {
         notifyAll();
```

Comment:
This `notifyAll()` should wake up the `wait(timetoTimeout);`.
Any idea why it doesn't work? Maybe you can explain your change a little more.